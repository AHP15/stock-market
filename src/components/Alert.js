import PropTypes from 'prop-types';

import styles from '../styles/Alert.module.css';

// eslint-disable-next-line arrow-body-style
const Alert = ({ message, close }) => {
  return (
    <div className={styles.alert}>
      <p>{message}</p>
      <button type="button" onClick={close}>x</button>
    </div>
  );
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  close: PropTypes.func.isRequired,
};

export default Alert;
