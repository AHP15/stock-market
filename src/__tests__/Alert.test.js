import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import Alert from '../components/Alert';

it('renders Alert correctly', () => {
  const tree = renderer.create(<Alert message="test" close={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
