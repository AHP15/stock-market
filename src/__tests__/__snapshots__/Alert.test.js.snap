// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Alert correctly 1`] = `
<div
  className="alert"
>
  <p>
    test
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    x
  </button>
</div>
`;
